### PR TITLE
[madmoizelle] Add new extractor

### PR DESCRIPTION
**Pull request by crazyiop**
solve issues #6491

### CODE REVIEW

File: youtube_dl/extractor/madmoizelle.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from __future__ import unicode_literals
+
+import re
+<<<<<<< HEAD
+from ..utils import (
+    HEADRequest,
+)
+=======
+from ..utils import HEADRequest
+>>>>>>> 7a50899fde5f12ad3f44bd92141a1161a139f0ee
```
Comment (TingPing) [inline youtube_dl/extractor/madmoizelle.py]: Leftovers from a merge.

File: youtube_dl/extractor/madmoizelle.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title')
+        description = self._html_search_meta('description', webpage)
+
+        formats = []
+        for format in re.finditer('jwplayer.*"(?P<url>[0-9]+.*(?P<quality>hd|sd).+)"', webpage):
+            url = 'https://player.vimeo.com/external/' + format.group('url')
+
+            # the url found point to a header redirection that we must follow
```
Comment (dstftw) [inline youtube_dl/extractor/madmoizelle.py]: What's the point? Let downloader follow it when downloading happens.

File: youtube_dl/extractor/madmoizelle.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title')
+        description = self._html_search_meta('description', webpage)
+
+        formats = []
+        for format in re.finditer('jwplayer.*"(?P<url>[0-9]+.*(?P<quality>hd|sd).+)"', webpage):
+            url = 'https://player.vimeo.com/external/' + format.group('url')
```
Comment (dstftw) [inline youtube_dl/extractor/madmoizelle.py]: Do not shadow `url` variable.

File: youtube_dl/extractor/madmoizelle.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+from ..utils import (
+    HEADRequest,
+)
+
+from .common import InfoExtractor
+
+
+class MadmoizelleIE(InfoExtractor):
+    IE_NAME = 'madmoizelle.com'
+    IE_DESC = 'madmoizelle JW player'
```
Comment (dstftw) [inline youtube_dl/extractor/madmoizelle.py]: Either remove or change to official site name.

File: youtube_dl/extractor/madmoizelle.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    _VALID_URL = r'(https?://)?(?:www\.)?madmoizelle\.com/.+-(?P<id>[0-9]+)#?.*'
```
Comment (dstftw) [inline youtube_dl/extractor/madmoizelle.py]: `https?://` should be mandatory.

**Comment by dstftw** [comment]
Multiple videos may be embedded: http://www.madmoizelle.com/apocalypse-verdun-502961#gs.s5Ztwk0.

File: youtube_dl/extractor/wat.py [hunk 6]
```diff
@@ -1,137 +0,0 @@
...
-            'id': real_id,
-            'display_id': display_id,
-            'title': first_chapter['title'],
-            'thumbnail': first_chapter['preview'],
-            'description': first_chapter['description'],
-            'view_count': video_info['views'],
-            'upload_date': upload_date,
-            'duration': first_file['duration'],
-            'formats': formats,
-        }
```
Comment (dstftw) [inline youtube_dl/extractor/wat.py]: I guess this was deleted by a mistake.

**Comment by yan12125** [comment]
Since #8497 landed, please move changes in ```youtube_dl/extractors/__init__.py``` to ```youtube_dl/extractors/extractors.py```. Check step 5 of the new [developer instructions](https://github.com/rg3/youtube-dl/blob/master/README.md#adding-support-for-a-new-site) for more information.

**Pull request closed without merging**